Sources/ArcGISToolkit/Components/Scalebar/ScalebarViewModel.swift

```diff
@@ -56,8 +56,10 @@ final class ScalebarViewModel: ObservableObject {
         let numberString = numberFormatter.string(
             from: NSNumber(value: altMapLength)
         ) ?? ""
-        let bottomUnitsText = " \(altDisplayUnits.abbreviation)"
-        let label = "\(numberString)\(bottomUnitsText)"
+        let label = String.totalLengthLabel(
+            length: numberString,
+            unitLabel: altDisplayUnits.localizedAbbreviation
+        )
```

Comment:
Have you considered using a [formatted measurement](https://developer.apple.com/documentation/foundation/measurement#3820415) as the label? That would simplify this code and might reduce the number of localized strings needed in the toolkit.

Reply:
That's what I was thinking as well. I thought I recalled we had a way to go from Esri Linear Units to Measurements and localizing Measurements could be done by iOS.

Reply:
We do. They're not in the documentation as DocC doesn't currently support documenting members of types in a different module, but we have `class UnitAngle.fromAngularUnit(_:)`, `class UnitArea.fromAreaUnit(_:)`, and `class UnitLength.fromLinearUnit(_:)` for this very purpose.

Reply:
Implement - its a much cleaner approach, thank you both.

Reply:
🥇

Reply:
There's a reason I referred you to the `formatted` methods on `Measurement` itself: "When working in Swift, use `formatted` methods directly on the types you want to format" (see [Data Formatting](https://developer.apple.com/documentation/foundation/data_formatting)). That's what we should do. For starters, add a convenience initializer to `Measurement`:

```swift
extension Measurement where UnitType == UnitLength {
    init(value: Double, linearUnit: LinearUnit) {
        self.init(value: value, unit: .fromLinearUnit(linearUnit))
    }
}
```

Then instead of having a method on `String` to format the values, just add a reusable format style:

```swift
extension FormatStyle where Self == Measurement<UnitLength>.FormatStyle {
    static var scaleMeasurement: Self {
        .measurement(width: .abbreviated, usage: .asProvided)
    }
}
```

FYI: That `measurement` method takes an optional `numberFormatStyle` that can be used to customize the format of the number part of the measurement.

Then, use those at the point of use:

```swift
// In 'ScalebarViewModel.getter:alternateUnit'

let measurement = Measurement(value: altMapLength, linearUnit: altDisplayUnits)
let label = measurement.formatted(.scaleMeasurement)

// In 'ScalebarViewModel.updateLabels()'

let segmentText: String
if index == numSegments - 1, let displayUnit {
    let measurement = Measurement(
        value: segmentMapLength,
        linearUnit: displayUnit
    )
    segmentText = measurement.formatted(.scaleMeasurement)
} else {
    segmentText = segmentMapLength.formatted(.number)
}
```

Reply:
Yes thank you. I processed the feedback from your second comment but then forgot to process your first. Thank you for the suggestions. Done now.